nucypher/utilities/gas_strategies.py

```diff
@@ -64,23 +63,43 @@ def datafeed_fallback_gas_price_strategy(web3: Web3, transaction_params: TxParam
 }
 
 
-def wrap_web3_gas_strategy(web3_gas_strategy: Callable):
+def wrap_web3_gas_strategy(speed: Optional[str] = None):
     """
     Enriches the web3 exceptions thrown by gas strategies
     """
-    @functools.wraps(web3_gas_strategy)
+    web3_gas_strategy = __RAW_WEB3_GAS_STRATEGIES[speed]
+
     def _wrapper(*args, **kwargs):
         try:
             return web3_gas_strategy(*args, **kwargs)
         except ValidationError as e:
             raise GasStrategyError("Calling the web3 gas strategy failed, probably due to an unsynced chain.") from e
```

Comment:
Helpful - but hopefully never encountered if stalecheck middleware is injected.

Reply:
🤞

Reply:
It's also very presumptuous, and contingent on web3 middleware configuration to say *" probably due to an unsynced chain."*.

"Probably" is pretty deflating in terms of helping the end-user (try top offer direction help).  How about:

"Verify your ethereum provider connection and syncing status"

Reply:
✅